megalinter/descriptors/lua.megalinter-descriptor.yml

```
@@ -54,3 +57,37 @@ linters:
       vscode:
         - name: vscode-luacheck
           url: https://marketplace.visualstudio.com/items?itemName=dwenegar.vscode-luacheck
+
+  # SELENE
+  - linter_name: selene
+    linter_url: https://kampfkarren.github.io/selene/
+    linter_repo: https://github.com/Kampfkarren/selene
+    linter_rules_url: https://kampfkarren.github.io/selene/lints/index.html
+    linter_rules_configuration_url: https://kampfkarren.github.io/selene/usage/configuration.html
+    linter_rules_inline_disable_url: https://kampfkarren.github.io/selene/usage/filtering.html
+    linter_megalinter_ref_url: "no"
+    config_file_name: selene.toml
+    cli_config_arg_name: "--config"
+    examples:
+      - "selene myfile.lua"
+      - "selene --config chktexrc.toml myfile.lua"
+    install:
+      apk:
+        - g++
+      dockerfile:
+        - |
+          RUN cargo install --branch main --git https://github.com/Kampfkarren/selene selene
```

I just remembered we have some handling for cargo crates directly

```suggestion
      apk:
        - g++
      cargo:
        - selene
```

Or if g++ isn't needed anymore

```suggestion
      cargo:
        - selene
```


If we support the version number correctly:

```suggestion
      cargo:
        - selene@0.27.1
```

---

Indeed, if there is a single cargo package in the dockerfile, rust toolchain commands are automatically added before :)

---

@nvuillam  Do we support having docker env vars as the cargo version when building our Dockerfiles? Like

```yaml
cargo:
  - selene@${LUA_SELENE_VERSION}
dockerfile:
  - |-
    # renovate: datasource=crate depName=selene
    ARG LUA_SELENE_VERSION=0.27.1
```

Would the arg be placed before using the cargo to install?